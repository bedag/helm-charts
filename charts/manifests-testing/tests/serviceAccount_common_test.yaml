suite: Test ServiceAccount Common Values
templates:
  - manifests.yaml
tests:
##
## Passing Test Scenarios
##

  ##
  ##
  - it: Common Labels & Partial Name
    values:
      - ./values/common/name.yaml
      - ./values/common/labels.yaml
    set:
      unit:
        manifests: 
          serviceaccount:
            enabled: true
            create: true
            labels:
              "test.label/1": true
    asserts:
      - equal:
          path: metadata.name
          value: "RELEASE-NAME-partial"
      - equal:
          path: metadata.labels.[app.kubernetes.io/instance]
          value: "RELEASE-NAME"
      - equal:
          path: metadata.labels.[app.kubernetes.io/name]
          value: "RELEASE-NAME-partial"
      - equal:
          path: metadata.labels.[app.kubernetes.io/instance]
          value: "RELEASE-NAME"
      - equal:
          path: metadata.labels.[test.label/1]
          value: true
      - isNotEmpty:
          path: metadata.labels.[app.kubernetes.io/version] 
      - isNotEmpty:
          path: metadata.labels.[app.kubernetes.io/managed-by] 

  ##
  ##
  - it: Local Name has priority over NameOverwrite
    values:
      - ./values/common/name.yaml
    set:
      unit:
        manifests: 
          serviceaccount:
            enabled: true
            create: true
            name: "overwrite"
    asserts:
      - equal:
          path: metadata.name
          value: "overwrite"

  ##
  ##
  - it: Local Name has priority over FullnameOverwrite
    values:
      - ./values/common/fullname.yaml
    set:
      unit:
        manifests: 
          serviceaccount:
            enabled: true
            create: true
            name: "overwrite"
    asserts:
      - equal:
          path: metadata.name
          value: "overwrite"

  ##
  ##
  - it: Fullname Overwrite
    values:
      - ./values/common/fullname.yaml
      - ./values/common/labels.yaml
    set:
      unit:
        manifests: 
          serviceaccount:
            enabled: true
            create: true
    asserts:
      - equal:
          path: metadata.name
          value: "fullname"
      - equal:
          path: metadata.labels.[app.kubernetes.io/name]
          value: "fullname"
      - equal:
          path: metadata.labels.[app.kubernetes.io/instance]
          value: "RELEASE-NAME"
      - equal:
          path: metadata.labels.[app.kubernetes.io/instance]
          value: "RELEASE-NAME"
      - isNotEmpty:
          path: metadata.labels.[app.kubernetes.io/version] 
      - isNotEmpty:
          path: metadata.labels.[app.kubernetes.io/managed-by] 

  ##
  ##
  - it: Fullname and Labels Overwrite
    values:
      - ./values/common/fullname.yaml
      - ./values/common/overwrites.yaml
    set:
      unit:
        manifests: 
          serviceaccount:
            enabled: true
            create: true
    asserts:
      - equal:
          path: metadata.name
          value: "fullname"
      - equal:
          path: metadata.labels.[common.overwrite/1]
          value: true
      - isEmpty:
          path: metadata.labels.[app.kubernetes.io/instance]
      - isEmpty:
          path: metadata.labels.[app.kubernetes.io/version] 
      - isEmpty:
          path: metadata.labels.[app.kubernetes.io/managed-by] 


  ##
  ##
  - it: Merge Global PullSecrets
    values:
      - ./values/common/fullname.yaml
      - ./values/common/overwrites.yaml
    set:
      unit:
        manifests: 
          serviceaccount:
            enabled: true
            create: true
            imagePullSecrets:
              - account-local-secret-1
              - account-local-secret-2
            globalPullSecrets: true
    asserts:
      - equal:
          path: metadata.name
          value: "fullname"
      - equal:
          path: metadata.labels.[common.overwrite/1]
          value: true
      - isEmpty:
          path: metadata.labels.[app.kubernetes.io/instance]
      - isEmpty:
          path: metadata.labels.[app.kubernetes.io/version] 
      - isEmpty:
          path: metadata.labels.[app.kubernetes.io/managed-by] 


      imagePullSecrets: []
      globalPullSecrets: false

