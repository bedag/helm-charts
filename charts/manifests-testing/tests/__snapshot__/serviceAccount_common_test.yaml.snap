Full Configured (Snapshot):
  1: |
    apiVersion: v1beta1
    automountServiceAccountToken: true
    kind: ServiceAccount
    metadata:
      annotations:
        test.annotations/1: value
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME-manifests-testing
        app.kubernetes.io/version: 0.0.0
        test.label/1: value
      name: RELEASE-NAME-manifests-testing
      namespace: custom-namespace
    secrets:
    - name: build-robot-token-bvbk5
